azurelinuxagent/common/protocol/wire.py

```diff
@@ -1150,37 +1247,40 @@ def get_header_for_cert(self):
 
     def get_host_plugin(self):
         if self._host_plugin is None:
-            goal_state = GoalState.fetch_goal_state(self)
+            goal_state = GoalState.fetch_goal_state(self, self.ext_config_retriever)
             self._set_host_plugin(HostPluginProtocol(self.get_endpoint(),
                                                      goal_state.container_id,
                                                      goal_state.role_config_name))
         return self._host_plugin
 
     def has_artifacts_profile_blob(self):
-        ext_conf = self.get_ext_conf()
-        return ext_conf and not \
-            textutil.is_str_none_or_whitespace(ext_conf.artifacts_profile_blob)
+        return self._goal_state is not None and not \
+            textutil.is_str_none_or_whitespace(self._goal_state.artifacts_profile_blob_url)
 
-    def get_artifacts_profile_through_host(self, blob):
+    def get_artifacts_profile_through_host(self, blob, etag=None):
         host = self.get_host_plugin()
-        uri, headers = host.get_artifact_request(blob)
-        profile = self.fetch(uri, headers, use_proxy=False)
-        return profile
+        uri, headers = host.get_artifact_request(blob, etag=etag)
+        profile, etag, not_modified = self.fetch_content_and_etag(uri, headers, use_proxy=False)
```

Comment:
NIT: Same as above, change variable name from `not_modified` to `modified`

Reply:
see comment above